src/RoslynPad.Common.UI/ViewModels/DocumentViewModel.cs

```diff
@@ -65,24 +80,24 @@ public static string GetAutoSaveName(string name)
             return name + AutoSaveSuffix + DefaultFileExtension;
         }
 
-        public static DocumentViewModel CreateRoot(string rootPath)
+        public static DocumentViewModel CreateRoot(string rootPath, DocumentFileWatcher documentFileWatcher)
         {
-            return new DocumentViewModel(rootPath);
+            IOUtilities.PerformIO(() => Directory.CreateDirectory(rootPath));
+            return new DocumentViewModel(rootPath, documentFileWatcher);
         }
 
-        public static DocumentViewModel FromPath(string path)
+        public static DocumentViewModel FromPath(string path, DocumentFileWatcher documentFileWatcher)
```

Comment:
I think the watcher can be removed in all usages of this method (see comments in other files).